framework-cloud/framework-git/src/main/java/org/kie/cloud/git/gogs/GogsGitProvider.java

```diff
@@ -0,0 +1,93 @@
+package org.kie.cloud.git.gogs;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.apache.http.entity.ContentType;
+import org.kie.cloud.git.AbstractGitProvider;
+import org.apache.http.client.fluent.Request;
+import org.kie.cloud.git.constants.GitConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GogsGitProvider extends AbstractGitProvider {
+
+    private String url;
+    private String user;
+    private String password;
+
+    private static final String URL_API_SUFFIX = "/api/v1";
+    private static final String URL_CREATE_REPOSITORY_SUFFIX = "/user/repos";
+    private static final String URL_REPOSITORY_SUFFIX = "/repos";
+
+    private static final Logger logger = LoggerFactory.getLogger(GogsGitProvider.class);
+
+    @Override public String createGitRepositoryWithPrefix(String repositoryPrefixName, String repositoryPath) {
+        final String repositoryName = generateRepositoryName(repositoryPrefixName);
+        createRepository(repositoryName);
+        pushToGitRepository(getRepositoryUrl(repositoryName), repositoryPath,
+                user, password);
+
+        return repositoryName;
+    }
+
+    @Override public void deleteGitRepository(String repositoryName) {
+        try {
+            final int statusCode = Request.Delete(url + URL_API_SUFFIX + URL_REPOSITORY_SUFFIX + "/" + user + "/" +
```

Comment:
Building URLs using string concatenation can be error prone (e.g. what if some piece ends/doesn't end with '/'?). Unfortunately I'm not aware of java ecosystem that solves this in a nicer way :P